docs/operations/hook-sidecar.md

```diff
@@ -0,0 +1,162 @@
+# Hook Sidecar Container
+
+## Introduction
+
+In KubeVirt, a Hook Sidecar container is a sidecar container (a secondary container that runs along with the main
+application container within the same Pod) used to apply customizations before the Virtual Machine is initialized. This
+ability is provided since 100% of the [libvirt domain XML](https://libvirt.org/formatdomain.html) elements and
+attributes are not configurable/customizable using the VMI specification.
+
+The sidecar containers communicate with the main container over a socket with a gRPC protocol. There are two main
+sidecar hooks:
+1. `onDefineDomain`: This hook helps to customize libvirt's XML and return the new XML over gRPC for the VM creation.
+2. `preCloudInitIso`: This hook helps to customize the cloud-init configuration. It operates on and returns JSON
+   formatted cloud-init data.
```

Comment:
This did not break down as a list in the end result page. I guess it misses an empty line between the list and previous paragraph.

Reply:
Thanks for the catch. Fixed.